### PR TITLE
Set the executable bit before running the tests

### PR DESCRIPTION
**Pull request by adithyaov**
Looks like cabal-sdist along with tar togather with some symphony
of design + bugs don't preserve the executable bit on
files. Manually setting the executable bit before running the
tests solves the issue.

### CODE REVIEW

**Comment by harendra-kumar** [comment]
Why this problem was not caught is because of this line in the CI:

```
      DISABLE_SDIST_BUILD: "y"
```

**Comment by harendra-kumar** [comment]
cabal sdist strips the executable bits deliberately. See https://github.com/haskell/cabal/issues/6514 for some discussion about it.

File: test/Streamly/System/Process.hs [hunk 1]
```diff
@@ -448,6 +449,10 @@ processChunks'4 = monadicIO $ run checkFailAction
 
 main :: IO ()
 main = do
+    -- Make sure all the executable files have the executable bit set before
+    -- running the tests.
+    for_ [executableFile, executableFileFail, executableFilePass]
+        $ \f -> S.drain $ Proc.toChunks "chmod" ["+x", f]
```
Comment (harendra-kumar) [inline test/Streamly/System/Process.hs:455]: We should not be making changes to the source tree for running tests. We can just remove the execute permissions on these files and run them using the shell like `sh <filename` or `/usr/bin/env sh <filename>`.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: test/Streamly/System/Process.hs [hunk 2]
```diff
@@ -157,7 +166,8 @@ toBytes2 = monadicIO $ run checkFailAction
     where
 
     action = do
-        S.drain $ Proc.toBytes' executableFileFail []
+        S.drain $
+             Proc.toBytes' interpreterFile [interpreterArg, executableFileFail]
```
Comment (harendra-kumar) [inline test/Streamly/System/Process.hs:170]: There is a lot of repetition, we can perhaps use an idiom like we have in streamly-shell.

**Review by harendra-kumar — APPROVED** [review]

File: test/Streamly/System/Process.hs (hunk 2 above)
Comment (adithyaov) [reply test/Streamly/System/Process.hs:170]: Probably. We can probably abstract stuff out later.

**Pull request merged**